File: agent/modules/email.py
# agent/modules/email.py
import hashlib
import re

import dns.resolver
import requests


def module_email(target: str) -> dict:
    result = {"target": target}

    if not re.match(r"^[^@\s]+@[^@\s]+\.[^@\s]+$", target):
        return {"error": "invalid email syntax"}

    local, domain = target.split("@", 1)
    result["local"] = local
    result["domain"] = domain

    try:
        answers = dns.resolver.resolve(domain, "MX", lifetime=5)
        result["mx"] = sorted([str(r.exchange).rstrip(".") for r in answers])
    except Exception as e:
        result["mx"] = []
        result["mx_error"] = str(e)

    try:
        answers = dns.resolver.resolve(domain, "A", lifetime=5)
        result["a_records"] = [str(r) for r in answers]
    except Exception:
        result["a_records"] = []

    try:
        h = hashlib.md5(target.strip().lower().encode()).hexdigest()
        r = requests.get(
            f"https://www.gravatar.com/avatar/{h}?d=404",
            timeout=8,
            headers={"User-Agent": "Mozilla/5.0"},
        )
        if r.status_code == 200:
            result["gravatar"] = True
        elif r.status_code == 404:
            result["gravatar"] = False
        else:
            result["gravatar"] = None
            result["gravatar_http"] = r.status_code
    except Exception as e:
        result["gravatar"] = None
        result["gravatar_error"] = str(e)

    return result

File: panel/modules/renders/email.js
// panel/modules/renders/email.js
import { field } from "../utils.js";

export function renderEmailResult(r) {
    const rows = [];
    rows.push(field("Email", r.target));
    rows.push(field("Логин", r.local));
    rows.push(field("Домен", r.domain));
    rows.push(field("MX", r.mx?.length ? r.mx.join(", ") : "— нет", !r.mx?.length ? "bad" : "good"));
    rows.push(field("A-записи", r.a_records?.length ? r.a_records.join(", ") : "— нет", !r.a_records?.length ? "bad" : "good"));
    if (r.gravatar === true) rows.push(field("Gravatar", "✅ есть", "good"));
    else if (r.gravatar === false) rows.push(field("Gravatar", "❌ нет", "bad"));
    else rows.push(field("Gravatar", "⚠ не удалось проверить", "warn"));
    if (r.gravatar_error) rows.push(field("Ошибка Gravatar", r.gravatar_error, "warn"));
    return `<table class="kv">${rows.join("")}</table>`;
}